app/client/keybase/vault_test.go

```diff
@@ -0,0 +1,217 @@
+package keybase
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"testing"
+
+	"github.com/ory/dockertest"
+	"github.com/ory/dockertest/docker"
+	"github.com/pokt-network/pocket/shared/crypto"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	vk *vaultKeybase
+)
+
+func setupTestVaultKeybase(address string) (*vaultKeybase, error) {
+	return NewVaultKeybase(vaultKeybaseConfig{
+		Address: address,
+		Token:   "dev-only-token",
+		Mount:   "secret",
+	})
+}
+
+func TestMain(m *testing.M) {
+	// uses a sensible default on windows (tcp/http) and linux/osx (socket)
+	pool, err := dockertest.NewPool("")
+	if err != nil {
+		log.Fatalf("Could not construct pool: %s", err)
+	}
+
+	// uses pool to try to connect to Docker
+	err = pool.Client.Ping()
+	if err != nil {
+		log.Fatalf("Could not connect to Docker: %s", err)
+	}
+
+	// pulls an image, creates a container based on it and runs it
+	resource, err := pool.RunWithOptions(&dockertest.RunOptions{
+		Repository: "vault",
+		Tag:        "latest",
+		Env: []string{
+			"VAULT_DEV_ROOT_TOKEN_ID=dev-only-token",
+			"VAULT_DEV_LISTEN_ADDRESS=0.0.0.0:8200",
+		},
+		ExposedPorts: []string{"8200/tcp"},
+		PortBindings: map[docker.Port][]docker.PortBinding{
+			"8200/tcp": {
+				{HostIP: "127.0.0.1", HostPort: "8200"},
+			},
+		},
+		CapAdd: []string{"IPC_LOCK"},
+	}, func(config *docker.HostConfig) {
+		// set AutoRemove to true so that stopped container goes away by itself
+		config.AutoRemove = true
+		config.RestartPolicy = docker.RestartPolicy{
+			Name: "no",
+		}
+	})
+
+	if err != nil {
+		log.Fatalf("Could not start resource: %s", err)
+	}
+
+	err = resource.Expire(120) // Tell docker to hard kill the container in 2 minutes
+	if err != nil {
+		log.Fatalf("Could not set expiration on resource: %s", err)
+	}
+
+	// exponential backoff-retry, because the application in the container might not be ready to accept connections yet
+	if err = pool.Retry(func() error {
+
+		// get the port from the container
+		endpoint := resource.GetHostPort("8200/tcp")
+
+		// test connection to vault
+		vk, err = setupTestVaultKeybase(fmt.Sprintf("http://%s", endpoint))
+
+		if err != nil {
+			return err
+		}
+
+		_, err = vk.client.Sys().Health()
+
+		return err
+	}); err != nil {
+		log.Fatalf("Could not connect to docker: %s", err)
+	}
+
+	code := m.Run()
+
+	// You can't defer this because os.Exit doesn't care for defer
+	if err := pool.Purge(resource); err != nil {
+		log.Fatalf("Could not purge resource: %s", err)
+	}
+
+	os.Exit(code)
+}
+
+func TestVaultKeybase(t *testing.T) {
+
+	if vk == nil {
+		t.Fatalf("vk is nil")
+	}
+
+	// Test Create
+	_, err := vk.Create("passphrase", "hint")
+	if err != nil {
```

Comment:
Just a thought these could be simplified to something like
```suggestion
	require.NoError(t, err)
```

Would reduce all the if statments

Reply:
sry missed this earlier, great suggestion.